fineract-savings/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java

```diff
@@ -549,10 +549,10 @@ public void postInterest(final MathContext mc, final LocalDate interestPostingUp
                     if (interestEarnedToBePostedForPeriod.isGreaterThanOrEqualTo(Money.zero(currency))) {
 
                         newPostingTransaction = SavingsAccountTransaction.interestPosting(this, office(), interestPostingTransactionDate,
-                                interestEarnedToBePostedForPeriod, interestPostingPeriod.isUserPosting());
+                                interestEarnedToBePostedForPeriod, interestPostingPeriod.isUserPosting(), ExternalId.empty());
```

Comment:
We should check whether enable-auto-generated-external-id is true or false...

if it is true, we need to generate automatically external id.